fpm/src/fpm_targets.f90

```diff
@@ -72,61 +69,71 @@ subroutine targets_from_sources(model,sources)
         xsuffix = ''
     end if
 
-    with_lib = any([(sources(i)%unit_scope == FPM_SCOPE_LIB,i=1,size(sources))])
+    with_lib = any([((model%packages(j)%sources(i)%unit_scope == FPM_SCOPE_LIB, &
+                      i=1,size(model%packages(j)%sources)) &
+                      ,j=1,size(model%packages))])
```

Comment:
For consistency with the rest of the code base the comma should be trailing not leading here
```suggestion
                      i=1,size(model%packages(j)%sources)), &
                      j=1,size(model%packages))])
```